anti-analysis/anti-debugging/debugger-detection/check-for-trap-flag-exception.yml

```diff
@@ -4,7 +4,8 @@ rule:
     namespace: anti-analysis/anti-debugging/debugger-detection
     authors:
       - michael.hunhoff@mandiant.com
-    scope: basic block
+    scopes:
+      static: basic block
```

Comment:
@mr-tz @yelhamer

Reply:
an additional benefit is that we can potentially merge only a subset of migrated rules, those with explicit `scopes.dynamic` set, and consider any rules without this to be needing a migration. then we can perform the migration in phases, perhaps easy rules first and harder rules manually.

Reply:
meaning all rules need explicit `static` and `dynamic` scope definitions?

Reply:
all rules should have them, or the linter should complain, and indicate that the migration for that rule hasn't been completed yet.

Reply:
I am not opposed to this syntax, but I prefer the current syntax (missing dynamic == invalid for dynamic) since it feels more straight-forward. But as I said I am open t both if you wish to switch to that instead.

As for picking only a subset of rules that have the `scopes.dynamic` set, I think that it wouldn't be very beneficial since we currently don't have any custom-made dynamic rules for the new syntax, but rather only migrated rules. If we want to proceed with the rules migration gradually then I suggest we pick rules that have only a static scope, and add the other ones later on.